README.md
# simple-converter-ver2

Jest to druga wersja prostego przelicznika pikseli na procenty i odwrotnie. Wersja ta powstała, aby można było korzystać z przelicznika na telefonach, tablecie oraz laptopie.

Użyte technologie:
HTML, JQuery, SCSS/CSS



There is second (other) version of simple coefficient on percents pikseli and vice versa. This version emerge, in order to it was possible to use with coefficient on cell (cellular phone), tablecie and laptop. 

Used technologies:
HTML, JQuery, SCSS/CSS 


history.txt
21 stycznia 2019 - wersja 0.0.1
Rozpoczęcie nowej wersji kalkulatora wartośc z pikseli na procenty i odwrotnie, która  ma zapewnić lepszą responsywność, przelicznika.

22 stycznia 2019- wersja 0.0.2
Poprawki dla szerokości ekranu 320px.

23 stycznia 2019 - wersja 0.0.3
Dodanie zdarzeń dla pozostałych rozdzielczości.
  Optymalizacja kodu.

24 stycznia 2019 - wersja 0.0.4
Dodanie tłumaczenia strony z polskiego na angielski ver. 1.

26 stycznia 2019 - wersja 0.0.5
Tłumaczenie strony z polskiego na angielski ver. 2.

27 stycznia 2019 - wersja 0.0.6
Dopasowano wygląd konwertera dla szerokości 1024px (dla szerokości tabletu cavion 10 3 grq).

28 stycznia 2019 - wersja 0.0.7
Poprawka wyświetlania konwertera dla szerokości od 1024px.

29 stycznia 2019 - wersja 0.0.8
Zmieniono kolory przycisków.

30 stycznia 2019 - wersja 0.0.9
Dodano animacje ładowania strony w postaci liczby procentów.

1 luty 2019 - wersja 0.1.0
Dodano ukrywane/pokazywanie menu w zależności od szerokości ekranu (coś ala hamburger menu).


js/language.js
//obsługa zdarzeń dla przycisku 800x600 pikseli
let workingFile=require("./workingFile.js");

module.exports.language = function(){

  $("#language").on("click", ()=>{
        if($("#language").text() == "EN"){
        $(".title").empty().append("Coefficient of width and heights on percents");
        $("#pixelsToPercents").empty().append("Conversion of width: pixels to percents for width 800px ");
        $("#subTitlePixels").empty().append("It serve number (pixels)");
        $("#subTitlePercents").empty().append("It serve number (percents)");
        $("#calculatePixelsWidth").empty().append("calculate");
        $("#calculatePercentsWidth").empty().append("calculate");
        $("#scoreTitle").empty().append("Result");

        $("#pixelsToPercentsHeight").empty().append("Conversion of height: pixels to percents for height 600px ");
        $("#subTitlePixelsHeight").empty().append("It serve number (pixels)");
        $("#subTitlePercentsHeight").empty().append("It serve number (percents)");
        $("#calculatePixelsHeight").empty().append("calculate");
        $("#calculatePercentsHeight").empty().append("calculate");
        $("#scoreTitleHeight").empty().append("Result");


        $("#language").empty().append("PL");
      }else if($("#language").text() == "PL"){
        $(".title").empty().append("Przelicznik szerokości i wysokości na procenty");

        $("#pixelsToPercents").empty().append("Conversion of width: pixels to percents for width 800px ");
        $("#subTitlePixels").empty().append("Podaj liczbę (w pikselach)");
        $("#subTitlePercents").empty().append("Podaj liczbę (w procentach)");
        $("#calculatePixelsWidth").empty().append("oblicz");
        $("#calculatePercentsWidth").empty().append("Oblicz");
        $("#scoreTitle").empty().append("Wynik to:");

        $("#pixelsToPercentsHeight").empty().append("Kowersja wysokości z pikseli na procenty dla wysokości 600px");
        $("#subTitlePixelsHeight").empty().append("Podaj liczbę (w pikselach)");
        $("#subTitlePercentsHeight").empty().append("Podaj liczbę (w procentach)");
        $("#calculatePixelsHeight").empty().append("oblicz");
        $("#calculatePercentsHeight").empty().append("oblicz");
        $("#scoreTitleHeight").empty().append("Wynik to:");
        $("#language").empty().append("EN");
      }
  });
}


js/workingFile.js
let first=require("./screen800x600.js");
let second=require("./screen1024x768.js");
let third=require("./screen1280x768.js");
let fourth=require("./screen1360x768.js");
let fifth=require("./screen1366x768");

document.addEventListener("DOMContentLoaded", () => {
	if( $(window).width() < 1024){
		$("#showHide").hide();
		$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayNone");
		var percents = 0;
			var id = setInterval(frame, 15);
			function frame() {
				if (percents == 100) {
					clearInterval(id);
						$("#loading").hide();
						$("#showHide").show();
						$( window ).resize(function() {
							if($(window).width() < 1024){
								$("#showHide").show();
								$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayNone");
							}else{
								$("#showHide").hide();
								$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayBlock");
							}
						});
				} else {
					percents++;
					$("#loading").empty().append(percents + "%");
				}
			}
	}else if( $(window).width() >= 1024){
		$("#showHide").hide();
		$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayNone");
		var percents = 0;
			var id = setInterval(frame, 15);
			function frame() {
				if (percents == 100) {
					clearInterval(id);
						$("#loading").hide();
						$("#showHide").hide();
						$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayBlock");
				} else {
					percents++;
					$("#loading").empty().append(percents + "%");
				}
			}
	}

	function english(number3, number4){
		$("#left").append("<p id='pixelsToPercents'>Conversion of width: pixels to percents for width " + number3 +"px</p>");
		$("#left").append("<p id='subTitlePixels'>It serve number (pixels)</p>");
		$("#left").append("<input type='number' id='numberPixelsValue'>");
		$("#left").append("<button id='calculatePixelsWidth'>calculate</button>");
		$("#left").append("<p id='subTitlePercents'>It serve number (percents)</p>");
		$("#left").append("<input type='number' id='numberPercentsValue'>");
		$("#left").append("<button id='calculatePercentsWidth'>calculate</button>");
		$("#left").append("<p id='scoreTitle'>Result</p>");
		$("#left").append("<p id='scoreWidth' class='height5vh'></p>");

		$("#right").append("<p id='pixelsToPercentsHeight'>Conversion of height: pixels to percents for height "+number4+"px</p>");
		$("#right").append("<p id='subTitlePixelsHeight'>It serve number (pixels)</p>");
		$("#right").append("<input type='number' id='numberPixelsValueHeight'>");
		$("#right").append("<button id='calculatePixelsHeight'>calculate</button>");
		$("#right").append("<p id='subTitlePercentsHeight'>It serve number (percents)</p>");
		$("#right").append("<input type='number' id='numberPercentsValueHeight'>");
		$("#right").append("<button id='calculatePercentsHeight'>calculate</button>");
		$("#right").append("<p id='scoreTitleHeight'>Result</p>");
		$("#right").append("<p id='scoreHeight' class='height5vh'></p>");
	}

	function polish(number3, number4){
		$("#left").append("<p id='pixelsToPercents'>Konwersja szerokości z pikseli na procenty dla szerokości " + number3 +"px</p>");
		$("#left").append("<p id='subTitlePixels'>Podaj liczbę (w pikselach)</p>");
		$("#left").append("<input type='number' id='numberPixelsValue'>");
		$("#left").append("<button id='calculatePixelsWidth'>oblicz</button>");
		$("#left").append("<p id='subTitlePercents'>Podaj liczbę (w procentach)</p>");
		$("#left").append("<input type='number' id='numberPercentsValue'>");
		$("#left").append("<button id='calculatePercentsWidth'>oblicz</button>");
		$("#left").append("<p id='scoreTitle'>Wynik to:</p>");
		$("#left").append("<p id='scoreWidth' class='height5vh'></p>");

		$("#right").append("<p id='pixelsToPercentsHeight'>Konwersja wysokości z pikseli na procenty dla wysokości "+number4+"px</p>");
		$("#right").append("<p id='subTitlePixelsHeight'>Podaj liczbę (w pikselach)</p>");
		$("#right").append("<input type='number' id='numberPixelsValueHeight'>");
		$("#right").append("<button id='calculatePixelsHeight'>oblicz</button>");
		$("#right").append("<p id='subTitlePercentsHeight'>Podaj liczbę (w procentach)</p>");
		$("#right").append("<input type='number' id='numberPercentsValueHeight'>");
		$("#right").append("<button id='calculatePercentsHeight'>oblicz</button>");
		$("#right").append("<p id='scoreTitleHeight'>Wynik to:</p>");
		$("#right").append("<p id='scoreHeight' class='height5vh'></p>");
	}

	$("#language").on("click", ()=>{
		if( $("#language").text() == "EN"){
			$("#right, #left").empty();
			$("#language").empty().append("PL");
			$(".title").empty().append("Coefficient of width and heights on percents");
		}else if( $("#language").text() == "PL"){
			$("#right, #left").empty();
			$("#language").empty().append("EN");
			$(".title").empty().append("Przelicznik szerokości i wysokości na procenty");
		}
	});

$("#showHide").on("click", ()=>{
				$("#first, #second, #third, #fourth, #fifth, #language").toggleClass("displayBlock");
				if($('#showHide').text() == "+"){
					$('#showHide').text("-");
				}else{
					$('#showHide').text("+");
				}
	});

module.exports.xxx = function(number1, number2, number3, number4){
  $("#right, #left").empty();

	if($("#language").text() == "EN"){
	  polish(number3, number4);
//width part
		$("#calculatePixelsWidth").on("click", () =>{
			let widthPixels = $("#numberPixelsValue").val();
			$("#scoreWidth").text(widthPixels/number1+ "%");
		});

		$("#calculatePercentsWidth").on("click", ()=>{
			let widthPercents = $("#numberPercentsValue").val();
			$("#scoreWidth").text(widthPercents*number1 + "px");
		});

	//height part
		$("#calculatePixelsHeight").on("click", () =>{
			let heightPixels = $("#numberPixelsValueHeight").val();
			$("#scoreHeight").text(heightPixels/number2 + "%");
		});

		$("#subTitlePercentsHeight").text("Podaj liczbę (w procentach)");
		$("#calculatePercentsHeight").on("click", ()=>{
			let heightPercents = $("#numberPercentsValueHeight").val();
			$("#scoreHeight").text(heightPercents*number2 + "px");
		});
	}else if($("#language").text() == "PL"){
		english(number3, number4);

	//width part
		$("#calculatePixelsWidth").on("click", () =>{
			let widthPixels = $("#numberPixelsValue").val();
			$("#scoreWidth").text(widthPixels/number1+ "%");
		});

		$("#calculatePercentsWidth").on("click", ()=>{
			let widthPercents = $("#numberPercentsValue").val();
			$("#scoreWidth").text(widthPercents*number1 + "px");
		});

	//height part
		$("#calculatePixelsHeight").on("click", () =>{
			let heightPixels = $("#numberPixelsValueHeight").val();
			$("#scoreHeight").text(heightPixels/number2 + "%");
		});

		$("#subTitlePercentsHeight").text("Podaj liczbę (w procentach)");
		$("#calculatePercentsHeight").on("click", ()=>{
			let heightPercents = $("#numberPercentsValueHeight").val();
			$("#scoreHeight").text(heightPercents*number2 + "px");
		});
	}
}
  first.first();
  second.second();
  third.third();
  fourth.fourth();
  fifth.fifth();

});//koniec DOMContentLoaded


js/engine.js
/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 1);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {

"use strict";


var first = __webpack_require__(2);
var second = __webpack_require__(3);
var third = __webpack_require__(4);
var fourth = __webpack_require__(5);
var fifth = __webpack_require__(6);

document.addEventListener("DOMContentLoaded", function () {
	if ($(window).width() < 1024) {
		var frame = function frame() {
			if (percents == 100) {
				clearInterval(id);
				$("#loading").hide();
				$("#showHide").show();
				$(window).resize(function () {
					if ($(window).width() < 1024) {
						$("#showHide").show();
						$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayNone");
					} else {
						$("#showHide").hide();
						$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayBlock");
					}
				});
			} else {
				percents++;
				$("#loading").empty().append(percents + "%");
			}
		};

		$("#showHide").hide();
		$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayNone");
		var percents = 0;
		var id = setInterval(frame, 15);
	} else if ($(window).width() >= 1024) {
		var _frame = function _frame() {
			if (percents == 100) {
				clearInterval(id);
				$("#loading").hide();
				$("#showHide").hide();
				$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayBlock");
			} else {
				percents++;
				$("#loading").empty().append(percents + "%");
			}
		};

		$("#showHide").hide();
		$("#first, #second, #third, #fourth, #fifth, #language").addClass("displayNone");
		var percents = 0;
		var id = setInterval(_frame, 15);
	}

	function english(number3, number4) {
		$("#left").append("<p id='pixelsToPercents'>Conversion of width: pixels to percents for width " + number3 + "px</p>");
		$("#left").append("<p id='subTitlePixels'>It serve number (pixels)</p>");
		$("#left").append("<input type='number' id='numberPixelsValue'>");
		$("#left").append("<button id='calculatePixelsWidth'>calculate</button>");
		$("#left").append("<p id='subTitlePercents'>It serve number (percents)</p>");
		$("#left").append("<input type='number' id='numberPercentsValue'>");
		$("#left").append("<button id='calculatePercentsWidth'>calculate</button>");
		$("#left").append("<p id='scoreTitle'>Result</p>");
		$("#left").append("<p id='scoreWidth' class='height5vh'></p>");

		$("#right").append("<p id='pixelsToPercentsHeight'>Conversion of height: pixels to percents for height " + number4 + "px</p>");
		$("#right").append("<p id='subTitlePixelsHeight'>It serve number (pixels)</p>");
		$("#right").append("<input type='number' id='numberPixelsValueHeight'>");
		$("#right").append("<button id='calculatePixelsHeight'>calculate</button>");
		$("#right").append("<p id='subTitlePercentsHeight'>It serve number (percents)</p>");
		$("#right").append("<input type='number' id='numberPercentsValueHeight'>");
		$("#right").append("<button id='calculatePercentsHeight'>calculate</button>");
		$("#right").append("<p id='scoreTitleHeight'>Result</p>");
		$("#right").append("<p id='scoreHeight' class='height5vh'></p>");
	}

	function polish(number3, number4) {
		$("#left").append("<p id='pixelsToPercents'>Konwersja szerokości z pikseli na procenty dla szerokości " + number3 + "px</p>");
		$("#left").append("<p id='subTitlePixels'>Podaj liczbę (w pikselach)</p>");
		$("#left").append("<input type='number' id='numberPixelsValue'>");
		$("#left").append("<button id='calculatePixelsWidth'>oblicz</button>");
		$("#left").append("<p id='subTitlePercents'>Podaj liczbę (w procentach)</p>");
		$("#left").append("<input type='number' id='numberPercentsValue'>");
		$("#left").append("<button id='calculatePercentsWidth'>oblicz</button>");
		$("#left").append("<p id='scoreTitle'>Wynik to:</p>");
		$("#left").append("<p id='scoreWidth' class='height5vh'></p>");

		$("#right").append("<p id='pixelsToPercentsHeight'>Konwersja wysokości z pikseli na procenty dla wysokości " + number4 + "px</p>");
		$("#right").append("<p id='subTitlePixelsHeight'>Podaj liczbę (w pikselach)</p>");
		$("#right").append("<input type='number' id='numberPixelsValueHeight'>");
		$("#right").append("<button id='calculatePixelsHeight'>oblicz</button>");
		$("#right").append("<p id='subTitlePercentsHeight'>Podaj liczbę (w procentach)</p>");
		$("#right").append("<input type='number' id='numberPercentsValueHeight'>");
		$("#right").append("<button id='calculatePercentsHeight'>oblicz</button>");
		$("#right").append("<p id='scoreTitleHeight'>Wynik to:</p>");
		$("#right").append("<p id='scoreHeight' class='height5vh'></p>");
	}

	$("#language").on("click", function () {
		if ($("#language").text() == "EN") {
			$("#right, #left").empty();
			$("#language").empty().append("PL");
			$(".title").empty().append("Coefficient of width and heights on percents");
		} else if ($("#language").text() == "PL") {
			$("#right, #left").empty();
			$("#language").empty().append("EN");
			$(".title").empty().append("Przelicznik szerokości i wysokości na procenty");
		}
	});

	$("#showHide").on("click", function () {
		$("#first, #second, #third, #fourth, #fifth, #language").toggleClass("displayBlock");
		if ($('#showHide').text() == "+") {
			$('#showHide').text("-");
		} else {
			$('#showHide').text("+");
		}
	});

	module.exports.xxx = function (number1, number2, number3, number4) {
		$("#right, #left").empty();

		if ($("#language").text() == "EN") {
			polish(number3, number4);
			//width part
			$("#calculatePixelsWidth").on("click", function () {
				var widthPixels = $("#numberPixelsValue").val();
				$("#scoreWidth").text(widthPixels / number1 + "%");
			});

			$("#calculatePercentsWidth").on("click", function () {
				var widthPercents = $("#numberPercentsValue").val();
				$("#scoreWidth").text(widthPercents * number1 + "px");
			});

			//height part
			$("#calculatePixelsHeight").on("click", function () {
				var heightPixels = $("#numberPixelsValueHeight").val();
				$("#scoreHeight").text(heightPixels / number2 + "%");
			});

			$("#subTitlePercentsHeight").text("Podaj liczbę (w procentach)");
			$("#calculatePercentsHeight").on("click", function () {
				var heightPercents = $("#numberPercentsValueHeight").val();
				$("#scoreHeight").text(heightPercents * number2 + "px");
			});
		} else if ($("#language").text() == "PL") {
			english(number3, number4);

			//width part
			$("#calculatePixelsWidth").on("click", function () {
				var widthPixels = $("#numberPixelsValue").val();
				$("#scoreWidth").text(widthPixels / number1 + "%");
			});

			$("#calculatePercentsWidth").on("click", function () {
				var widthPercents = $("#numberPercentsValue").val();
				$("#scoreWidth").text(widthPercents * number1 + "px");
			});

			//height part
			$("#calculatePixelsHeight").on("click", function () {
				var heightPixels = $("#numberPixelsValueHeight").val();
				$("#scoreHeight").text(heightPixels / number2 + "%");
			});

			$("#subTitlePercentsHeight").text("Podaj liczbę (w procentach)");
			$("#calculatePercentsHeight").on("click", function () {
				var heightPercents = $("#numberPercentsValueHeight").val();
				$("#scoreHeight").text(heightPercents * number2 + "px");
			});
		}
	};
	first.first();
	second.second();
	third.third();
	fourth.fourth();
	fifth.fifth();
}); //koniec DOMContentLoaded

/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

module.exports = __webpack_require__(0);


/***/ }),
/* 2 */
/***/ (function(module, exports, __webpack_require__) {

"use strict";


//obsługa zdarzeń dla przycisku 800x600 pikseli
var workingFile = __webpack_require__(0);

module.exports.first = function () {
  $("#first").on("click", function () {
    workingFile.xxx(8, 6, 800, 600);
  });
};

/***/ }),
/* 3 */
/***/ (function(module, exports, __webpack_require__) {

"use strict";


//obsługa zdarzeń dla przycisku 1024x768 pikseli
var workingFile = __webpack_require__(0);

module.exports.second = function () {

    $("#second").on("click", function () {
        workingFile.xxx(10.24, 7.68, 1024, 768);
    });
};

/***/ }),
/* 4 */
/***/ (function(module, exports, __webpack_require__) {

"use strict";


//obsługa zdarzeń dla przycisku 800x600 pikseli
var workingFile = __webpack_require__(0);

module.exports.third = function () {

  $("#third").on("click", function () {
    workingFile.xxx(12.80, 7.68, 1280, 768);
  });
};

/***/ }),
/* 5 */
/***/ (function(module, exports, __webpack_require__) {

"use strict";


//obsługa zdarzeń dla przycisku 800x600 pikseli
var workingFile = __webpack_require__(0);

module.exports.fourth = function () {

  $("#fourth").on("click", function () {
    workingFile.xxx(1360, 7.68, 1360, 768);
  });
};

/***/ }),
/* 6 */
/***/ (function(module, exports, __webpack_require__) {

"use strict";


//obsługa zdarzeń dla przycisku 800x600 pikseli
var workingFile = __webpack_require__(0);

module.exports.fifth = function () {

  $("#fifth").on("click", function () {
    workingFile.xxx(1366, 7.68, 1366, 768);
  });
};

/***/ })
/******/ ]);